### PR TITLE
Fullstack/seats to generated

### PR DESCRIPTION
**Pull request by aslakhol**
The core idea of this PR is to move away from composite key in seats and rather use a primary key generated by the DB. 

Why? Part of the composite key we used was available for the user to edit. The user has some expectations about how they can change text fields and they don't line up with the strict restrictions of a database PK. 

The PR include changes to the backend to actually change the model and how the rest of the backend interacts with it, along with updated tests. 

Since the model changed it no longer made any sense that the seats are accessed by roomId + what is now called seat name. As a result the endpoints are rewritten so that they use the new id. The new structure also made it possible to update seats without having to delete and recreate seats each time. So an endpoint is made for this. 

And then there is a major refactor of the frontend to use these new endpoints. 
There are updates to the calls, but also to much of the surrounding logic. Since we now rely on an ID generated by the database we had to do some thinking to maintain local state for seats. 

Lastly there was an issue with two ISeats definitions that had to be resolved. 

This PR should also make it easier to solve #262, #238 and #247

### CODE REVIEW

File: backend/models/room.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Room(db.Model):
     seats = db.relationship(
         'Seat',
         back_populates="room",
-        foreign_keys=[Seat.room_id, Seat.seat_id],
+        foreign_keys=[Seat.room_id, Seat.id],
```
Comment (s0lvang) [inline backend/models/room.py]: I think you can remove the foreign_keys-field. Look at how it's done in other relations

File: backend/services/userService.py [hunk 2]
```diff
@@ -40,7 +40,12 @@ def deleteUser(id):
         db.session.delete(user)
         db.session.commit()
         return user.to_json(), 200
-    except(SQLAlchemyError):
+    except SQLAlchemyError as err:
+        print(err, flush=True)
+        return "", 400
+    except AssertionError as err:
+        print('her er jeg', flush=True)
```
Comment (s0lvang) [inline backend/services/userService.py]: dummy

Comment (s0lvang) [reply backend/services/userService.py]: Do you even backend? Fucking frontend developer

File: frontend/src/components/Seats/index.tsx [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+import { IPostSeat } from '../../API/interfaces';
 import { IStore } from '../../store';
 import { createSeatAction, deleteSeatAction, updateSeatAction } from './actions';
 import Presentational from './Presentational';
 
 export interface ISeat {
-  id: string;
+  id: number;
   info: string;
+  name: string;
```
Comment (FluidSense) [inline frontend/src/components/Seats/index.tsx]: This does not reflect seat in backend, seat_name is nullable and could thus be undefined

File: frontend/src/components/Seats/actions.ts [hunk 4]
```diff
@@ -20,26 +21,23 @@ const seatDeleted = (success: boolean) => ({
...
     else dispatch(seatDeleted(false));
   };
 
-export const updateSeatAction = (roomId: number, oldSeatId: string, newSeat: ISeat):
+export const updateSeatAction = (seatId: number, newName: string):
   ThunkAction<Promise<void>, {}, {}, AnyAction> => async (dispatch: Dispatch) => {
-    const deleteResults = await deleteSeat(roomId, oldSeatId);
-    if (deleteResults) {
-      const createResult = await postSeat(newSeat);
-      if (createResult) dispatch(seatUpdated(true));
```
Comment (FluidSense) [inline frontend/src/components/Seats/actions.ts]: On `seatUpdated`, would it perhaps be an idea to update seats in `state.rooms.rooms` as well, so store reflects backend?

Comment (aslakhol) [reply frontend/src/components/Seats/actions.ts]: I've created an issue for this and we'll deal with it after merging this PR. #304

**Comment by FluidSense** [comment]
Names doesn't seem to be set properly. I created seat B1, B2, A3 on a room, saved, and got B1, A2, A3.

**Comment by FluidSense** [comment]
Removing student from seat through rooms no longer work with this PR, ID is never sent. Results in 500 Internal server error "Int object has no attribute get"

**Comment by FluidSense** [comment]
Assign seat through application not possible, it never turns active

**Comment by aslakhol** [comment]
> Names doesn't seem to be set properly. I created seat B1, B2, A3 on a room, saved, and got B1, A2, A3.

Could you be more explicit? I can't reproduce this.

**Comment by FluidSense** [comment]
> >  Names doesn't seem to be set properly. I created seat B1, B2, A3 on a room, saved, and got B1, A2, A3.  

> Could you be more explicit? I can't reproduce this.  

Me neither, I guess I must have been tired.

File: frontend/src/components/AdminViewRoom/SeatPlacer.tsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ class SeatPlacer extends React.Component<IProps, IState> {
 
     if (seat.user) {
       return (
-        <Panel border={true} style={{ display:'flex', width:'35%' }}>
+        <Panel border={true} style={{ display: 'flex', width: '35%' }}>
           <PanelBase border={true}>{seat.id}</PanelBase>
```
Comment (FluidSense) [inline frontend/src/components/AdminViewRoom/SeatPlacer.tsx]: {seat.id} should be {seat.name}

**Pull request merged**